src/main/java/cinema/security/AuthenticationServiceImpl.java

```diff
@@ -1,25 +1,31 @@
 package cinema.security;
 
 import cinema.exceptions.AuthenticationException;
-import cinema.lib.Inject;
-import cinema.lib.Service;
 import cinema.model.User;
 import cinema.service.ShoppingCartService;
 import cinema.service.UserService;
 import cinema.util.HashUtil;
 import java.util.Optional;
 import org.apache.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
 
 @Service
 public class AuthenticationServiceImpl implements AuthenticationService {
     private static final Logger logger = Logger.getLogger(AuthenticationServiceImpl.class);
-
-    @Inject
     private UserService userService;
-
-    @Inject
     private ShoppingCartService shoppingCartService;
 
+    @Autowired
+    public AuthenticationServiceImpl(UserService userService) {
```

Comment:
will both userService and shoppingCartService be initialized, try to test it in main for example just for yourself?

Reply:
done